process-engine/process_engine_runtime
# 🐛 Fix timer expression parsing

**Changes:**

1. Remove all timer expression validation from the ModelParser.
    - Doing this during Model-parsing had the unfortunate side effect of preventing the use of token-expressions.
2. Implement timer validation in the TimerFacade.
3. Move all token expression functions from the FlowNodeHandlers to the TimerFacade.
    - This way, token expression parsing and timer validation gets executed for all timers on all FlowNodeHandlers that use them.
4. Fix an `Unhandled Promise rejection` issue with the `IntermediateTimerEvent` handler, which could occur when an error was thrown from within the TimerFacade.
5. Remove some tests for the ProcessModelParser that have become obsolete with this change.
6. Add tests for the TimerFacade's timer validations.
 
**Issues:**

Closes #249

PR: #250

## How can others test the changes?

- Deploy a ProcessModel that contains a timer event with a timer-expression that makes use of the ProcessToken.
- Deployment should end successfully.
- Try running that ProcessModel.
- The timer should be parsed and executed correctly.

## PR-Checklist

Please check the boxes in this list after submitting your PR:
